sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2770,6 +2770,18 @@ class Dataset[T] private[sql](
    */
   def take(n: Int): Array[T] = head(n)
 
+  /**
+   * Returns the last `n` rows in the Dataset.
```

Comment:
Is this `last` only making sense for ordered dataset?

Though `head` also refers to first rows, it can be seen as same as `limit`. For `last` it seems to be more related to an ordered dataset?

Reply:
Yes, I rather targets the last rows (with natural order or explicit ordering) .. I think it's a similar argument. If the order is non-deterministic, there won't be not much difference comparing to `head()` .